File: api/src/main/java/dev/vibris/api/ContextApplyResult.java
package dev.vibris.api;

import java.util.Objects;

public record ContextApplyResult(boolean successful, SceneContext context, String message) {
    public ContextApplyResult {
        context = Objects.requireNonNull(context, "context");
        message = Objects.requireNonNull(message, "message");
    }

    public static ContextApplyResult success(SceneContext context) {
        return new ContextApplyResult(true, context, "");
    }

    public static ContextApplyResult failure(SceneContext context, String message) {
        return new ContextApplyResult(false, context, message);
    }
}

File: api/src/main/kotlin/dev/vibris/api/ContextApplyResult.kt
package dev.vibris.api

@JvmRecord
data class ContextApplyResult(
    val successful: Boolean,
    val context: SceneContext,
    val message: String,
) {
    companion object {
        @JvmStatic
        fun success(context: SceneContext): ContextApplyResult = ContextApplyResult(true, context, "")

        @JvmStatic
        fun failure(context: SceneContext, message: String): ContextApplyResult =
            ContextApplyResult(false, context, message)
    }
}